pkg/server/updates.go

```diff
@@ -97,10 +103,40 @@ type versionInfo struct {
 	Details string `json:"details"`
 }
 
+// PeriodicallyClearStmtStats is backstop -- generally the diagnostics reporting
```

Comment:
I'm really sorry but I don't understand this description comment.
When I start reading this I want to know:
- what this function does on the first line
- what is the mechanism by which it guarantees that no stats are lost via a reset if reporting does work (incidentally: I do not see any such mechanism in the implementation below)
- not too many sub-clauses in each sentence. The last sentence is particularly hard to parse.

Also incidentally I do not understand "it drives that reset"